services/requester/cloud_kms_rotation_signer.go

```diff
@@ -0,0 +1,104 @@
+package requester
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/onflow/flow-go-sdk/crypto"
+	"github.com/onflow/flow-go-sdk/crypto/cloudkms"
+	"github.com/rs/zerolog"
+)
+
+var _ crypto.Signer = &CloudKMSKeyRotationSigner{}
+
+// CloudKMSKeyRotationSigner is a crypto signer that contains a pool of
+// `crypto.Signer`[1] objects, each of which is tied to a Cloud KMS
+// asymmetric signing key.
+// It keeps track of the signer/key combination that should be used for
+// the next incoming signing request. This allows for faster submission
+// of transactions to the network, due to a sequence number not being
+// reused between different keys used.
+// It also contains logic to queue up signature requests and in case
+// there are more transactions pending than the keys in the pool it
+// will wait for transactions to get executed so the new sequence key
+// can be obtained.
+// The signer is concurrency-safe.
+//
+// [1](https://github.com/onflow/flow-go-sdk/blob/master/crypto/cloudkms/signer.go#L37)
+type CloudKMSKeyRotationSigner struct {
+	mux        sync.RWMutex
+	kmsSigners []*cloudkms.Signer
+	index      int
+	signersLen int
+	logger     zerolog.Logger
+}
+
+// NewSignerForKeys returns a new CloudKMSKeyRotationSigner, for the
+// given slice of Cloud KMS keys.
+func NewSignerForKeys(
```

Comment:
since the KMS integration test won't be possible looks like as part of the CI, let's add a unit test that mocks out KMS signer and tests the functioning of this struct

Reply:
This [comment](https://github.com/onflow/flow-evm-gateway/pull/282#discussion_r1633321987), suggested to move the creation of `cloudkms.Client` to `requester`, which is basically the `NewKMSKeyRotationSigner` constructor.
This means that we can no longer inject this dependency, in order to write a proper unit test.
Should I undo the suggestion of your first comment?